internal/agent/delta/store.go

```diff
@@ -87,6 +89,58 @@ type Store struct {
 	CacheDir string
 	// NextIDMap stores the information about the available plugins
 	NextIDMap pluginIDMap
+	// stores time of last success submission of inventory to backend
+	lastSuccessSubmission time.Time
+}
+
+func (d *Store) LastSuccessSubmission() (time.Time, error) {
+
+	if !d.lastSuccessSubmission.Equal(time.Time{}) {
+		return d.lastSuccessSubmission, nil
+	}
+
+	content, err := ioutil.ReadFile(d.generateLastSuccessSubmissionFilePath())
+
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	if err = d.lastSuccessSubmission.UnmarshalText(content); err != nil {
+		return time.Time{}, err
+	}
+
+	if !d.lastSuccessSubmission.Equal(time.Time{}) {
+		return d.lastSuccessSubmission, nil
+	}
+
+	return time.Time{}, ErrNoPreviousSuccessSubmissionTime
+}
+
+func (d *Store) UpdateAndSaveLastSuccessSubmission() error {
```

Comment:
There is any reasoning to have update and save separated?
There is any use case where we want to save w/o update or otherwise?

In my opinion update implies both (memory and disk) as a required to persist the value. 
I would merge code and/or rename the function to only Update

Reply:
there's an ongoing work to revamp this approach at https://github.com/newrelic/infrastructure-agent/compare/last-submission-time-persistence...last-submission-time-persistence-revamp